bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
Data ranges interrogate renderers for their preferred bounds, so I think its actually the converse. The current behavior is to try and give "nice" bounds. Although, its dependent on the quality of the bounding box returned by the renderer, and not all of them do an equally good job. 

"Strict" bounds is just a simple enveloping of all the data in some set of columns in a data source or sources. It's the way column data sources originally behaved, and the reason for hacks like `range_padding`. (Sometimes "range padding is useful, but it's not sufficient to implement "nice" bounds and thats what it what it was trying to be used for.) This behavior was replaced with the "ask the renderer" behavior some time ago. 

We could certainly add back a range that does simple enveloping, and then maybe an `AutoRange` that has a policy that delegates to one of the specific types (I would not want to dump the strict enveloping code back into the current data range)